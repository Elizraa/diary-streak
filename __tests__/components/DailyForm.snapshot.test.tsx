import DailyStreakForm from '@/components/DailyForm';
import { render } from '@testing-library/react';
import { act } from 'react';

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}));

it('renders Daily Streak Form unchanged', async () => {
  let container;

  await act(async () => {
    ({ container } = render(<DailyStreakForm />));
  });

  expect(container).toMatchSnapshot();
});
