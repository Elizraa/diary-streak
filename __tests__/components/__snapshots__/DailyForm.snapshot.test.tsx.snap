// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Daily Streak Form unchanged 1`] = `
<div>
  <div
    class="text-card-foreground flex flex-col gap-6 rounded-xl border py-6 w-full shadow-lg border-gray-800 bg-gray-900"
    data-slot="card"
  >
    <div
      class="@container/card-header grid auto-rows-min grid-rows-[auto_auto] items-start gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 bg-gradient-to-r from-gray-900 to-gray-800 rounded-t-lg border-b border-gray-800 pt-6 -mt-5"
      data-slot="card-header"
    >
      <div
        class="leading-none font-semibold text-purple-400"
        data-slot="card-title"
      >
        Record Your Daily Streak
      </div>
      <div
        class="text-sm text-gray-400"
        data-slot="card-description"
      >
        Enter your username and PIN to log today's stamp
      </div>
    </div>
    <div
      class="px-6 bg-gray-900"
      data-slot="card-content"
    >
      <form
        class="space-y-4"
      >
        <div
          class="space-y-2"
        >
          <label
            class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-gray-300"
            data-slot="label"
            for="username"
          >
            Username
          </label>
          <input
            class="file:text-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 flex h-9 w-full min-w-0 rounded-md border px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border-gray-700 bg-gray-800 text-white focus-visible:ring-purple-500 placeholder:text-gray-500"
            data-slot="input"
            id="username"
            placeholder="Enter your username"
            value=""
          />
        </div>
        <div
          class="space-y-2"
        >
          <label
            class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-gray-300"
            data-slot="label"
            for="pin"
          >
            PIN
          </label>
          <input
            class="file:text-foreground selection:bg-primary selection:text-primary-foreground dark:bg-input/30 flex h-9 w-full min-w-0 rounded-md border px-3 py-1 text-base shadow-xs transition-[color,box-shadow] outline-none file:inline-flex file:h-7 file:border-0 file:bg-transparent file:text-sm file:font-medium disabled:pointer-events-none disabled:cursor-not-allowed disabled:opacity-50 md:text-sm focus-visible:border-ring focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive border-gray-700 bg-gray-800 text-white focus-visible:ring-purple-500 placeholder:text-gray-500"
            data-slot="input"
            id="pin"
            inputmode="numeric"
            pattern="[0-9]*"
            placeholder="Enter your PIN"
            type="password"
            value=""
          />
        </div>
        <div
          class="space-y-2"
        >
          <label
            class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-gray-300"
            data-slot="label"
          >
            How are you feeling today?
             
            <span
              class="text-gray-500 text-sm"
            >
              (optional)
            </span>
          </label>
          <div
            class="flex gap-3"
          >
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-primary-foreground shadow-xs px-4 py-2 has-[>svg]:px-3 flex-1 h-16 bg-gray-800 hover:bg-gray-700 border border-gray-700"
              data-slot="button"
              data-testid="happy-mood"
              type="button"
            >
              <svg
                class="lucide lucide-smile h-8 w-8 text-gray-400"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <path
                  d="M8 14s1.5 2 4 2 4-2 4-2"
                />
                <line
                  x1="9"
                  x2="9.01"
                  y1="9"
                  y2="9"
                />
                <line
                  x1="15"
                  x2="15.01"
                  y1="9"
                  y2="9"
                />
              </svg>
            </button>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive text-primary-foreground shadow-xs px-4 py-2 has-[>svg]:px-3 flex-1 h-16 bg-gray-800 hover:bg-gray-700 border border-gray-700"
              data-slot="button"
              data-testid="sad-mood"
              type="button"
            >
              <svg
                class="lucide lucide-frown h-8 w-8 text-gray-400"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <path
                  d="M16 16s-1.5-2-4-2-4 2-4 2"
                />
                <line
                  x1="9"
                  x2="9.01"
                  y1="9"
                  y2="9"
                />
                <line
                  x1="15"
                  x2="15.01"
                  y1="9"
                  y2="9"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="space-y-2"
        >
          <label
            class="flex items-center gap-2 text-sm leading-none font-medium select-none group-data-[disabled=true]:pointer-events-none group-data-[disabled=true]:opacity-50 peer-disabled:cursor-not-allowed peer-disabled:opacity-50 text-gray-300"
            data-slot="label"
            for="pin"
          >
            Notes 
            <span
              class="text-gray-500 text-sm"
            >
              (optional)
            </span>
          </label>
          <textarea
            class="shadow-xs focus-visible:border-ring focus-visible:ring-[3px] border border-gray-700 bg-gray-800 rounded-xl text-white focus-visible:ring-purple-500 placeholder:text-gray-500 p-2 w-full h-32 resize-y"
            data-testid="note-textarea"
            id="note"
            placeholder="Is there a noteworthy event today, or what do you hope will happen?"
            rows="4"
            style="white-space: pre-wrap; overflow-wrap: break-word; text-align: left;"
          />
        </div>
        <button
          class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-all disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 shrink-0 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive shadow-xs h-9 px-4 py-2 has-[>svg]:px-3 w-full bg-purple-600 hover:bg-purple-700 text-white mt-6"
          data-slot="button"
          type="submit"
        >
          Submit Daily Stamp
        </button>
      </form>
    </div>
    <div
      class="items-center px-6 [.border-t]:pt-6 flex justify-center border-t border-gray-800 pt-4 bg-gray-900 rounded-b-lg"
      data-slot="card-footer"
    >
      <p
        class="text-sm text-gray-400 inline-flex items-center"
      >
        Keep up your daily stamp for commitment!
         
        <svg
          class="lucide lucide-book-open-check ml-1 text-orange-300"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 21V7"
          />
          <path
            d="m16 12 2 2 4-4"
          />
          <path
            d="M22 6V4a1 1 0 0 0-1-1h-5a4 4 0 0 0-4 4 4 4 0 0 0-4-4H3a1 1 0 0 0-1 1v13a1 1 0 0 0 1 1h6a3 3 0 0 1 3 3 3 3 0 0 1 3-3h6a1 1 0 0 0 1-1v-1.3"
          />
        </svg>
      </p>
    </div>
  </div>
</div>
`;
